File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        unordered_set<int> s;

        int m = nums[0];
        s.insert(nums[0]);
        int M = nums[1];
        s.insert(nums[1]);

        if (m > M) {
            swap (m, M);
        }

        for (int i=2; i<nums.size(); i++){
            if (nums[i] > M) {
                M = nums[i];
            } else if (nums[i] < m) {
                m = nums[i];
            }

            s.insert(nums[i]);
        }

        vector<int> ans;
        for (int i=m; i<=M; i++){
            if (s.find(i) == s.end()){
                ans.push_back(i);
            }
        }

        return ans;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        HashSet<Integer> set = new HashSet<>();

        int m = nums[0];
        int M = nums[1];

        if (m > M) {
            int t = m;
            m = M; 
            M = t;
        }

        set.add(m);
        set.add(M);

        for (int i=2; i<nums.length; i++){
            if (nums[i] < m) {
                m = nums[i];
            } else if (nums[i] > M) {
                M = nums[i];
            }

            set.add(nums[i]);
        }

        List<Integer> ans = new ArrayList<>();
        for (int i=m; i<=M; i++){
            if (!set.contains(i)) {
                ans.add(i);
            }
        }

        return ans;
    }
}